patches/server/1044-Add-Llama-caravan-events.patch

```diff
@@ -0,0 +1,98 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: David Scandurra <david.scandurra@check24.de>
+Date: Mon, 30 Oct 2023 03:41:35 +0100
+Subject: [PATCH] Add Llama caravan events
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/ai/goal/LlamaFollowCaravanGoal.java b/src/main/java/net/minecraft/world/entity/ai/goal/LlamaFollowCaravanGoal.java
+index 21725aee29e9120d1c7e1e19f91c21a73a28844f..778a4037f87a8cfe9be0c9ff9db12323cccbe0c2 100644
+--- a/src/main/java/net/minecraft/world/entity/ai/goal/LlamaFollowCaravanGoal.java
++++ b/src/main/java/net/minecraft/world/entity/ai/goal/LlamaFollowCaravanGoal.java
+@@ -61,7 +61,7 @@ public class LlamaFollowCaravanGoal extends Goal {
+             } else if (!llama.isLeashed() && !this.firstIsLeashed(llama, 1)) {
+                 return false;
+             } else {
+-                this.llama.joinCaravan(llama);
++                this.llama.joinCaravan(llama, io.papermc.paper.event.entity.LlamaJoinCaravanEvent.Reason.AI_GOAL); // Paper - add reason
+                 return true;
+             }
+         } else {
+@@ -97,7 +97,7 @@ public class LlamaFollowCaravanGoal extends Goal {
+ 
+     @Override
+     public void stop() {
+-        this.llama.leaveCaravan();
++        this.llama.leaveCaravan(io.papermc.paper.event.entity.LlamaLeaveCaravanEvent.Reason.LOST_HEAD); // Paper - add reason
+         this.speedModifier = 2.1D;
+     }
+ 
+diff --git a/src/main/java/net/minecraft/world/entity/animal/horse/Llama.java b/src/main/java/net/minecraft/world/entity/animal/horse/Llama.java
+index 5f61c97478f005aaaaad1b027118079db7275cf7..83ffa6b121460d769245c82337d06332e89a7289 100644
+--- a/src/main/java/net/minecraft/world/entity/animal/horse/Llama.java
++++ b/src/main/java/net/minecraft/world/entity/animal/horse/Llama.java
+@@ -437,8 +437,19 @@ public class Llama extends AbstractChestedHorse implements VariantHolder<Llama.V
+         }
+     }
+ 
++    // Paper start - Add Llama caravan events
+     public void leaveCaravan() {
```

Comment:
I think method should marked with `@Deprecated` and `@DoNotUse` to catch any future uses of it since they should all use the method with a reason.